ui/pages/confirmations/components/alerts/multiple-alert-modal/multiple-alert-modal.tsx

```diff
@@ -0,0 +1,107 @@
+import React, { useCallback, useMemo, useState } from 'react';
+import {
+  Box,
+  ButtonIcon,
+  ButtonIconSize,
+  IconName,
+  Text,
+} from '../../../../../components/component-library';
+import {
+  Display,
+  TextColor,
+  TextVariant,
+} from '../../../../../helpers/constants/design-system';
+import { useI18nContext } from '../../../../../hooks/useI18nContext';
+import useAlerts from '../../../hooks/useAlerts';
+import { AlertModal } from '../alert-modal';
+
+export type MultipleAlertModalProps = {
+  /** The unique identifier of the entity that owns the alert. */
+  ownerId: string;
+  /** The function to be executed when the button in the alert modal is clicked. */
+  handleButtonClick: () => void;
+  /** The unique key representing the specific alert field. */
+  alertKey: string;
+  /** The function to be executed when the modal needs to be closed. */
+  onClose: () => void;
+};
+
+export function MultipleAlertModal({
+  ownerId,
+  handleButtonClick,
+  alertKey,
+  onClose,
+}: MultipleAlertModalProps) {
+  const t = useI18nContext();
+  const { alerts } = useAlerts(ownerId);
+
+  const [selectedIndex, setSelectedIndex] = useState(
+    alerts.findIndex((alert) => alert.key === alertKey),
+  );
+
+  const selectedAlert = useMemo(
+    () => alerts[selectedIndex],
+    [alerts, selectedIndex],
+  );
+
+  const onBack = useCallback(() => {
+    setSelectedIndex((prevIndex) =>
+      prevIndex > 0 ? prevIndex - 1 : alerts.length - 1,
+    );
+  }, [alerts]);
+
+  const onNext = useCallback(() => {
+    setSelectedIndex((prevIndex) =>
+      prevIndex < alerts.length - 1 ? prevIndex + 1 : 0,
+    );
+  }, [alerts]);
+
+  const onButtonClick = useCallback(() => {
+    if (selectedIndex + 1 === alerts.length) {
+      handleButtonClick();
+      return;
+    }
+
+    onNext();
+  }, [handleButtonClick, onNext, selectedIndex, alerts]);
+
+  return (
+    <AlertModal
+      ownerId={ownerId}
+      handleButtonClick={onButtonClick}
+      alertKey={selectedAlert.key}
+      onClose={onClose}
+      multipleAlerts={
+        alerts.length > 1 ? (
```

Comment:
If this was in a separate function, we could early return here instead of nesting all the JSX?